src/services/HelperService.js

```diff
@@ -23,9 +23,14 @@ function * _checkRef (entity) {
   }
 
   if (entity.submissionId) {
-    const existSubmission = yield SubmissionService._getSubmission(entity.submissionId)
-
-    if (!existSubmission) {
+    const filter = {
+      TableName: 'Submission',
+      Key: {
+        'id': entity.submissionId
+      }
+    }
+    const existSubmission = yield dbhelper.getRecord(filter)
+    if (!existSubmission.Item) {
```

Comment:
I am not sure of this change - what was the issue with the earlier one? Is it that it returned reviews too?

Reply:
I'm assuming this func is here to check that the record we're calling for exists, across all the object types the API handles?

Reply:
The logic is to validate the existence of ID in the DynamoDB. Initially the function _getSubmission was doing that, now it has pieces of code to fetch the associated reviews and reviewSummations which is not required for existence check.

Hence to save on database calls, I made this change. Makes sense?

Reply:
Let's do it this way:
- Have an additional boolean parameter to the `_getSubmission` function that, if true, does not fetch the reviews and review summations for the submission.

Reply:
Can I say if the boolean parameter is false, don't fetch the reviews? The parameter naming and default value will be more meaningful, hence.

Reply:
Yeah that's ok. I figure using true will require less refactor effort